apps/mobile/src/screens/Onboarding/OnboardingProfileBioScreen.tsx

```diff
@@ -77,6 +97,72 @@ export function OnboardingProfileBioScreen() {
   if (user?.__typename !== 'GalleryUser') {
     throw new Error(`Unable to fetch the logged in user`);
   }
+  const web3Provider = useMemo(
+    () => (provider ? new ethers.providers.Web3Provider(provider) : undefined),
+    [provider]
+  );
+  const [isSigningIn, setIsSigningIn] = useState(false);
+  const handleDismiss = useCallback(() => {
+    provider?.disconnect();
+    setIsSigningIn(false);
+  }, [provider]);
+
+  const handleSignMessage = useCallback(async () => {
+    if (!web3Provider || !address) {
+      return;
+    }
+
+    const signer = web3Provider.getSigner();
+    const { nonce } = await createNonce(address, 'Ethereum');
+
+    try {
+      setIsSigningIn(true);
+      const signature = await signer.signMessage(nonce);
+
+      const { signatureValid } = await addWallet({
+        authMechanism: {
+          eoa: {
+            signature,
+            nonce,
+            chainPubKey: {
+              pubKey: address,
+              chain: 'Ethereum',
+            },
+          },
+        },
+        chainAddress: {
+          address,
+          chain: 'Ethereum',
+        },
+      });
+
+      if (!signatureValid) {
+        throw new Error('Signature is not valid');
+      }
+
+      if (!isSyncing) {
+        syncTokens('Ethereum');
+      }
+
+      bottomSheet.current?.dismiss();
+
+      navigation.navigate('OnboardingNftSelector', {
+        page: 'ProfilePicture',
+      });
```

Comment:
Is the increment token sync included when the existing user just added a new wallet? 

If yes, I can totally toss it later

Reply:
hm im not sure! lets discuss on slack